idea/idea-native/src/org/jetbrains/konan/analyser/KonanAnalyzerFacade.kt

```diff
@@ -61,66 +63,76 @@ class KonanAnalyzerFacade : ResolverForModuleFactory() {
 
     val konanPaths = KonanPaths.getInstance(project)
-    val libraryPaths = konanPaths.libraryPaths().toMutableList()
-    libraryPaths.addAll(konanPaths.konanPlatformLibraries())
 
     val moduleInfo = moduleContent.moduleInfo as? ModuleProductionSourceInfo
     val module = moduleInfo?.let { getModule(it) }
 
-    fun setLibraryForDescriptor(path: Path, descriptor: ModuleDescriptorImpl) {
+    fun setLibraryForDescriptor(libraryPath: File, descriptor: ModuleDescriptorImpl) {
       if (module == null) return
-      val library = findLibrary(module, path) ?: return
+      val library = findLibrary(module, libraryPath) ?: return
       //todo: replace reflection by changes to kotlin-native?
-      descriptor.setField("capabilities", { oldValue ->
+      descriptor.setField("capabilities") { oldValue ->
         val capabilities = oldValue as Map<*, *>
         val libraryInfo = createLibraryInfo(project, library)
         capabilities + Pair(ModuleInfo.Capability, libraryInfo)
-      })
+      }
     }
 
-    fun createDescriptor(path: Path): ModuleDescriptorImpl? {
-      val file = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(path.toFile())
-      return if (file != null && file.exists()) {
-        val descriptor = KonanDescriptorManager.INSTANCE.getDescriptor(file, languageVersionSettings)
-        setLibraryForDescriptor(path, descriptor)
-        descriptor
+    fun createDescriptor(libraryPath: File): ModuleDescriptorImpl? {
+      val virtualFile = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(libraryPath)
+      return if (virtualFile != null && virtualFile.exists()) {
+        val libraryDescriptor = KonanDescriptorManager.INSTANCE.getDescriptor(virtualFile, languageVersionSettings)
+        setLibraryForDescriptor(libraryPath, libraryDescriptor)
+        libraryDescriptor
       }
       else null
     }
 
-    val descriptors = libraryPaths.mapNotNull(::createDescriptor)
-    val stdlibDescriptor = konanPaths.konanStdlib()?.let(::createDescriptor)
+    val libraryDescriptors = mutableListOf<ModuleDescriptorImpl>()
+
+    if (module != null) {
+      ModuleRootManager.getInstance(module).orderEntries().forEachLibrary { library ->
+
+        val libraryPaths = library.getUrls(OrderRootType.CLASSES).map {
+          // cut off the schema ("jar:///" for ZIP files or "file:///" for directories)
+          // if this is a .klib file, then remove the rightmost part of the path that goes after the name of the file: "!/..."
+          File(URI(it).path.replaceAfterLast(".klib", ""))
+        }
+
+        libraryPaths.forEach { createDescriptor(it)?.also { libraryDescriptors.add(it) } }
+
+        true // continue the loop while there are more libs
+      }
+    }
 
-    val dependencies = descriptors.toMutableList()
-    if (stdlibDescriptor != null)
-      dependencies.add(stdlibDescriptor)
+    val stdlibDescriptor = konanPaths.konanStdlib()?.let { createDescriptor(it.toFile()) }
```

Comment:
At this point `libraryDescriptors` should already contain a descriptor for stdlib.

Reply:
rewritten in another way